kcp/utils.sh

```diff
@@ -167,3 +167,80 @@ registerSyncTarget() {
     echo "Waiting for the SyncTarget resource to reach ready state"
     KUBECONFIG="${CPS_KUBECONFIG}" kubectl wait --for=condition=Ready=true synctarget/"${WORKLOAD_CLUSTER}" --timeout 3m
 }
+
+createAPIBinding() {
+    exportName=$1
+    KUBECONFIG="${CPS_KUBECONFIG}" kubectl kcp ws
+    url=$(KUBECONFIG="${CPS_KUBECONFIG}" kubectl get workspace $SERVICE_WS -o jsonpath='{.status.URL}')
+    path=$(basename $url)
+    KUBECONFIG="${CPS_KUBECONFIG}" kubectl kcp ws use $USER_WS
+
+    permissionClaims='
+  permissionClaims:
+  - group: ""
+    resource: "secrets"
+    state: "Accepted"
+  - group: ""
+    resource: "namespaces"
+    state: "Accepted"'
+
+  if [ exportName == "gitopsrvc-appstudio-shared" ]; then
```

Comment:
What happens if  it is not `gitopsrvc-appstudio-shared`?

Reply:
This piece of the section is providing external permission claims required for exporting appstudio-shared APIExport, it is just a matter of fact we need to explicitly define `group: \"managed-gitops.redhat.com\"` `resource: \"gitopsdeployments\"` in this particular condition! If that's not the case, then function normally runs as defined.

```
  if [ exportName == "gitopsrvc-appstudio-shared" ]; then
    permissionClaims="${acceptedPermissionClaims}
  - group: \"managed-gitops.redhat.com\"
    resource: \"gitopsdeployments\"
    state: \"Accepted\"
    identityHash: ${identityHash}"
  fi
```